splunk-ingestable-s3-bucket/main.tf

```diff
@@ -0,0 +1,210 @@
+// S3 Bucket with a Lifecycle Policy that will delete items older that 7 days
+resource "aws_s3_bucket" "this" {
+  bucket = "${var.log_bucket_name}"
+  acl    = "private"
+
+  lifecycle_rule {
+    id      = "auto-delete-after-${var.expiration_days}-days"
+    prefix  = ""
+    enabled = true
+
+    expiration {
+      days = "${var.expiration_days}"
+    }
+  }
+
+  tags = {
+    Project     = "${var.project}"
+    Environment = "${var.environment}"
+  }
+}
+
+// Policy for the bucket with READ and LIST permissions given to the splunk account, Other accounts
+// Belonging to the organizaion are only allowed to Put objects to this bucket .
+
+resource "aws_s3_bucket_policy" "application_bucket_policy" {
+  bucket = "${aws_s3_bucket.this.id}"
+
+  policy = <<POLICY
+{
+   "Version": "2012-10-17",
+   "Statement": [
+      {
+         "Sid": "splunk_access",
+         "Effect": "Allow",
+         "Principal": {
+            "AWS" : "arn:aws:iam::${var.splunk_account_id}:root"
+         },
+         "Action": [
+            "s3:Get*",
+            "s3:List*"
+         ],
+         "Resource": [
+            "arn:aws:s3:::${var.log_bucket_name}"
+         ]
+      },
+      {
+         "Sid": "all_accounts_in_organization_can_put_object_access",
+         "Effect": "Allow",
+         "Principal": "*",
+         "Action": [
+            "s3:Put*"
+         ],
+         "Condition": {
+            "StringEquals":
+                {
+                "aws:PrincipalOrgID":["${var.aws_organization_id}"]
+                }
+         },
+         "Resource": [
+            "arn:aws:s3:::${var.log_bucket_name}/*"
+         ]
+      }
+   ]
+}
+POLICY
+}
+
+// S3 Bucket event notification that gets triggered on new items
+
+resource "aws_s3_bucket_notification" "object_created" {
+  bucket = "${aws_s3_bucket.this.id}"
+
+  topic {
+    topic_arn = "${aws_sns_topic.bucket_events.arn}"
+
+    events = [
+      "s3:ObjectCreated:*",
+    ]
+
+    filter_suffix = ""
+  }
+
+  depends_on = [
```

Comment:
not sure this `depends_on` is needed, because you are referring it in `topic` already.

Reply:
Probably right, I think I needed it previously because hard-coded ARNS was used and not references.

Reply:
👍 There are still edge cases with Terraform when `depends_on` should be used, but this is not the case.